test/integration/scheduler/main_test.go

```diff
@@ -17,11 +17,123 @@ limitations under the License.
 package scheduler
 
 import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"strings"
 	"testing"
 
+	"k8s.io/klog/v2/ktesting"
+	kubeschedulertesting "k8s.io/kubernetes/cmd/kube-scheduler/app/testing"
+	"k8s.io/kubernetes/test/integration"
 	"k8s.io/kubernetes/test/integration/framework"
 )
 
 func TestMain(m *testing.M) {
 	framework.EtcdMain(m.Run)
 }
+
+func TestHealthEndpoints(t *testing.T) {
+	server, apiserverConfig, _, err := integration.StartTestAPIServer(t)
```

Comment:
`"k8s.io/kubernetes/cmd/kube-apiserver/app/testing".StartTestServerOrDie`  could mock an apiserver.

Reply:
`StartTestAPIServer` is a wrapper of `"k8s.io/kubernetes/cmd/kube-apiserver/app/testing".StartTestServerOrDie` because we need to setup configs (eg. token, kubeconfig) and return them to caller for later uses.